osu.Game.Rulesets.Taiko/Difficulty/Skills/Stamina.cs

```diff
@@ -1,16 +1,48 @@
 ﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
 // See the LICENCE file in the repository root for full licence text.
 
-using System.Linq;
 using osu.Game.Rulesets.Difficulty.Preprocessing;
 using osu.Game.Rulesets.Difficulty.Skills;
-using osu.Game.Rulesets.Difficulty.Utils;
 using osu.Game.Rulesets.Mods;
 using osu.Game.Rulesets.Taiko.Difficulty.Preprocessing;
 using osu.Game.Rulesets.Taiko.Objects;
 
 namespace osu.Game.Rulesets.Taiko.Difficulty.Skills
 {
+    /// <summary>
+    /// Stamina of a single key, calculated based on repetition speed.
+    /// </summary>
+    public class SingleKeyStamina
+    {
+        private double previousHitTime = -1;
+
+        /// <summary>
+        /// Similar to <see cref="StrainDecaySkill.StrainValueOf(DifficultyHitObject)"/>
+        /// </summary>
+        public double StrainValueOf(DifficultyHitObject current)
+        {
+            if (previousHitTime == -1)
+            {
+                previousHitTime = current.StartTime;
+                return 0;
+            }
+
+            double objectStrain = 0.5;
+            objectStrain += speedBonus(current.StartTime - previousHitTime);
```

Comment:
I'm skimming the document linked in the OP, and it says this in the section pointing out issues with the current stamina setup:

> The base rewarding density is also an issue, with a base object strain of 1, density is rewarded at a fixed value rather than being able to calculate density per object at a differential.
The speed bonus is based on the duration of recent objects in the playfield, and thus added to the base object strain.

This doesn't seem to be far off from that, except that the values are slightly different now, which leaves me wondering - is this a temporary state of affairs, or is the doc outdated in some respect? Is this change supposed to improve things with respect to how strains are calculated?

Reply:
I'll update the document to the reflect the current changes now. Strain also is in the reworks of being rewritten, which is what I was referring to in the document and was kept out of this pr.